### PR TITLE
Fixed bug #7271

### PR DESCRIPTION
**Pull request by jlorente**
Fixed bug #7271. Classes that call '::populateRecord()' after retrieving data from ddbb call now the method on the class of the model returned by '::instantiate()'. Tests for mysql, elasticsearch and mongodb attached.

### CODE REVIEW

File: framework/db/ActiveQueryTrait.php [hunk 1]
```diff
@@ -129,12 +129,14 @@ private function createModels($rows)
             if ($this->indexBy === null) {
                 foreach ($rows as $row) {
                     $model = $class::instantiate($row);
+                    $class = get_class($model);
```
Comment (qiangxue) [inline framework/db/ActiveQueryTrait.php]: This is problematic as you are modifying `$class` which would be used in the next loop.

File: framework/db/ActiveQueryTrait.php [hunk 2]
```diff
@@ -129,12 +129,14 @@ private function createModels($rows)
...
                     $class::populateRecord($model, $row);
                     $models[] = $model;
                 }
             } else {
                 foreach ($rows as $row) {
                     $model = $class::instantiate($row);
+                    $class = get_class($model);
```
Comment (qiangxue) [inline framework/db/ActiveQueryTrait.php]: Same problem as above.

**Comment by qiangxue** [comment]
Everything else looks fine. Thanks.

**Comment by jlorente** [comment]
I have fixed the problem using a new variable.

**Comment by qiangxue** [comment]
Merged. Thanks!

**Pull request closed without merging**